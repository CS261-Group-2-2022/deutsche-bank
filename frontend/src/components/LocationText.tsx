import { LocationMarkerIcon } from "@heroicons/react/solid";

type LocationTextProps = {
  location: string;
  link?: string;
};

export default function LocationText({ location, link }: LocationTextProps) {
  return (
    <p className="flex text-gray-600 items-center">
      <LocationMarkerIcon className="mr-1 h-5 w-5" />
      {link ? (
        <a
          href={link}
          className="text-blue-600 underline underline-offset-1 visited:text-purple-600"
        >
          {location}
        </a>
      ) : (
        location
      )}
    </p>
  );
}
