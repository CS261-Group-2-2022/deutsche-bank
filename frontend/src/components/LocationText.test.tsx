import renderer from "react-test-renderer";
import LocationText from "./LocationText";

test("it renders location provided", () => {
  const component = renderer.create(<LocationText location="Building" />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("it renders link if provided", () => {
  const component = renderer.create(
    <LocationText location="Building" link="https://google.com" />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
