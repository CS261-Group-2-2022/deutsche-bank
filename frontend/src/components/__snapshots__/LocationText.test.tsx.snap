// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders link if provided 1`] = `
<p
  className="flex text-gray-600 items-center"
>
  <svg
    aria-hidden="true"
    className="mr-1 h-5 w-5"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M5.05 4.05a7 7 0 119.9 9.9L10 18.9l-4.95-4.95a7 7 0 010-9.9zM10 11a2 2 0 100-4 2 2 0 000 4z"
      fillRule="evenodd"
    />
  </svg>
  <a
    className="text-blue-600 underline underline-offset-1 visited:text-purple-600"
    href="https://google.com"
  >
    Building
  </a>
</p>
`;

exports[`it renders location provided 1`] = `
<p
  className="flex text-gray-600 items-center"
>
  <svg
    aria-hidden="true"
    className="mr-1 h-5 w-5"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M5.05 4.05a7 7 0 119.9 9.9L10 18.9l-4.95-4.95a7 7 0 010-9.9zM10 11a2 2 0 100-4 2 2 0 000 4z"
      fillRule="evenodd"
    />
  </svg>
  Building
</p>
`;
